web/packages/teleterm/src/ui/TabHost/TabHost.tsx

```diff
@@ -96,6 +97,14 @@ export function TabHost({
       onDuplicatePty: () => {
         documentsService.duplicatePtyAndActivate(doc.uri);
       },
+      onReopenPtyInShell(shell: Shell) {
+        if (
+          doc.kind === 'doc.terminal_shell' ||
+          doc.kind === 'doc.gateway_kube'
```

Comment:
Could we encapsulate this logic as `canDocChangeShell` in `web/packages/teleterm/src/ui/services/workspacesService/documentsService/types.ts`?

At the moment, it's repeated three times in `useDocumentTerminal.ts`, `tabContextMenu.ts` and here.

Reply:
Good idea, done.